math/generic/include/algebra/math/impl/generic_matrix.hpp

```diff
@@ -273,6 +273,62 @@ set_inplace_product_left_transpose(MA &A, const MB &B) requires(
   }
 }
 
+template <bool transpose, concepts::matrix M>
+ALGEBRA_HOST_DEVICE constexpr algebra::traits::value_t<M> transposable_get(
+    const M &A, const algebra::traits::index_t<M> i,
+    const algebra::traits::index_t<M> j) {
+  if constexpr (transpose) {
+    return algebra::traits::element_getter_t<M>()(A, j, i);
+  } else {
+    return algebra::traits::element_getter_t<M>()(A, i, j);
+  }
+}
+
+template <bool transpose_left, bool transpose_right, concepts::matrix MA,
+          concepts::matrix MB>
+ALGEBRA_HOST_DEVICE constexpr auto
+transposed_product(const MA &A, const MB &B) requires(
+    algebra::concepts::matrix_multipliable<
+        std::conditional_t<transpose_left,
+                           decltype(transpose(std::declval<MA>())), MA>,
+        std::conditional_t<transpose_right,
+                           decltype(transpose(std::declval<MB>())), MB> >) {
+  using index_t = algebra::traits::index_t<MA>;
+  using value_t = algebra::traits::value_t<MA>;
+
+  using effective_lhs_t =
+      std::conditional_t<transpose_left,
+                         decltype(transpose(std::declval<MA>())), MA>;
+  using effective_rhs_t =
+      std::conditional_t<transpose_right,
+                         decltype(transpose(std::declval<MB>())), MB>;
+
+  using result_t =
+      algebra::traits::get_matrix_t<MA, algebra::traits::rows<effective_lhs_t>,
+                                    algebra::traits::columns<effective_rhs_t>,
+                                    value_t>;
+
+  result_t C;
```

Comment:
```suggestion
  auto C = zero<result_t>();
```
Or is this slow, because it is another assignment, temporary etc?